autoPyTorch/optimizer/smbo.py

```diff
@@ -0,0 +1,316 @@
+            smac_scenario_args (typing.Optional[typing.Dict[str, typing.Any]]):
+                Additional arguments to the smac scenario
+            get_smac_object_callback (typing.Optional[typing.Callable]):
+                Allows to create a user specified SMAC object
+            ensemble_callback (typing.Optional[EnsembleBuilderManager]):
+                A callback used in this scenario to start ensemble building subtasks
+
+        """
+        super(AutoMLSMBO, self).__init__()
+        # data related
+        self.dataset_name = dataset_name
+        self.datamanager = None
+        self.metric = metric
+        self.task = None
+        self.backend = backend
+
+        # the configuration space
+        self.config_space = config_space
+
+        # the number of parallel workers/jobs
+        self.n_jobs = n_jobs
+        self.dask_client = dask_client
+
+        # Evaluation
+        self.resampling_strategy = resampling_strategy
+        if resampling_strategy_args is None:
+            resampling_strategy_args = {}
+        self.resampling_strategy_args = resampling_strategy_args
+
+        # and a bunch of useful limits
+        self.worst_possible_result = get_cost_of_crash(self.metric)
+        self.total_walltime_limit = int(total_walltime_limit)
+        self.func_eval_time_limit = int(func_eval_time_limit)
+        self.memory_limit = memory_limit
+        self.watcher = watcher
+        self.seed = seed
+        self.start_num_run = start_num_run
+        self.include = include
+        self.exclude = exclude
+        self.disable_file_output = disable_file_output
+        self.smac_scenario_args = smac_scenario_args
+        self.get_smac_object_callback = get_smac_object_callback
+
+        self.ensemble_callback = ensemble_callback
+
+        dataset_name_ = "" if dataset_name is None else dataset_name
+        logger_name = '%s(%d):%s' % (self.__class__.__name__, self.seed, ":" + dataset_name_)
+        self.logger = get_logger(logger_name)
+
+    def reset_data_manager(self) -> None:
+        if self.datamanager is not None:
+            del self.datamanager
+        if isinstance(self.dataset_name, AbstractDataManager):
+            self.datamanager = self.dataset_name
+        else:
+            self.datamanager = self.backend.load_datamanager()
+
+        self.task = self.datamanager.info['task']
+
+    def run_smbo(self, func: typing.Optional[typing.Callable] = None
+                 ) -> typing.Tuple[RunHistory, TrajLogger, str]:
+
+        self.watcher.start_task('SMBO')
+
+        # == first things first: load the datamanager
+        self.reset_data_manager()
+
+        # == Initialize non-SMBO stuff
+        # first create a scenario
+        seed = self.seed
+        self.config_space.seed(seed)
+        # allocate a run history
+        num_run = self.start_num_run
+
+        # Initialize some SMAC dependencies
+
+        if self.resampling_strategy in ['partial-cv',
```

Comment:
I think we wont have partial-cv or partial-cv-iterative-fit. It would be just cv